### PR TITLE
Fix issue of node.js binding for ContextOptions

### PR DESCRIPTION
**Pull request by mingmingtasd**
PTAL, thanks! @fujunwei

### CODE REVIEW

File: node/src/Context.cpp [hunk 1]
```diff
@@ -33,13 +33,15 @@ namespace node {
...
                 }
-                std::string powerPreferenceOption = optionsObject.Get("powerPreference").ToString();
-                if (powerPreferenceOption == "default") {
-                    options.powerPreference = MLPowerPreference_Default;
-                } else if (powerPreferenceOption == "low-power") {
-                    options.powerPreference = MLPowerPreference_Low_power;
-                } else if (powerPreferenceOption == "high-performance") {
-                    options.powerPreference = MLPowerPreference_High_performance;
+                std::string powerPreference = optionsObject.Get("powerPreference").ToString();
+                transform(powerPreference.begin(), powerPreference.end(), powerPreference.begin(),
```
Comment (fujunwei) [inline node/src/Context.cpp:37]: Add header file `#include <algorithm>` for `std::transform` and #include <cctype>for std::tolower. Add `std` prefixes before `transform `and `tolower`

File: node/src/Context.cpp [hunk 2]
```diff
@@ -53,14 +55,15 @@ namespace node {
...
-                    optionsObject.Get("devicePreference").ToString();
-                if (devicePreferenceOption == "default") {
-                    options.devicePreference = MLDevicePreference_Default;
-                } else if (devicePreferenceOption == "gpu") {
-                    options.devicePreference = MLDevicePreference_Gpu;
-                } else if (devicePreferenceOption == "cpu") {
-                    options.devicePreference = MLDevicePreference_Cpu;
+                std::string devicePreference = optionsObject.Get("devicePreference").ToString();
+                transform(devicePreference.begin(), devicePreference.end(),
+                          devicePreference.begin(), ::tolower);
```
Comment (fujunwei) [inline node/src/Context.cpp:60]: ditto.

File: node/src/Context.cpp (hunk 1 above)
Comment (mingmingtasd) [reply node/src/Context.cpp]: Updated, PTAL,thanks! @fujunwei

File: node/src/Context.cpp [hunk 3]
```diff
@@ -33,13 +35,16 @@ namespace node {
...
-                if (powerPreferenceOption == "default") {
-                    options.powerPreference = MLPowerPreference_Default;
-                } else if (powerPreferenceOption == "low-power") {
-                    options.powerPreference = MLPowerPreference_Low_power;
-                } else if (powerPreferenceOption == "high-performance") {
-                    options.powerPreference = MLPowerPreference_High_performance;
+                std::string powerPreference = optionsObject.Get("powerPreference").ToString();
+                std::transform(powerPreference.begin(), powerPreference.end(),
+                               powerPreference.begin(),
+                               [](char c) -> char { return std::tolower(c); });
```
Comment (huningxin) [inline node/src/Context.cpp:41]: I don't think the implementation should transform them to lowercase. If the string doesn't match the spec definition, it is a wrong type. If it is not set (or undefined), the implementation should use the default value.

File: node/src/Context.cpp [hunk 4]
```diff
@@ -53,14 +58,16 @@ namespace node {
...
-                if (devicePreferenceOption == "default") {
-                    options.devicePreference = MLDevicePreference_Default;
-                } else if (devicePreferenceOption == "gpu") {
-                    options.devicePreference = MLDevicePreference_Gpu;
-                } else if (devicePreferenceOption == "cpu") {
-                    options.devicePreference = MLDevicePreference_Cpu;
+                std::string devicePreference = optionsObject.Get("devicePreference").ToString();
+                std::transform(devicePreference.begin(), devicePreference.end(),
+                               devicePreference.begin(),
+                               [](char c) -> char { return std::tolower(c); });
```
Comment (huningxin) [inline node/src/Context.cpp:64]: ditto.

File: node/src/Context.cpp (hunk 1 above)
Comment (fujunwei) [reply node/src/Context.cpp]: A better catch is to assert the usage

https://github.com/webmachinelearning/webnn-native/blob/main/src/webnn_native/openvino/GraphIE.cpp#L852
https://github.com/webmachinelearning/webnn-native/blob/main/src/webnn_native/dml/GraphDML.cpp#L444
```
const char* deviceName = devicePreference == ml::DevicePreference::Cpu ||
                                         devicePreference == ml::DevicePreference::Default
                                     ? "CPU"
                                     : "GPU";
```
or in creating context https://github.com/webmachinelearning/webnn-native/blob/main/src/webnn_native/WebnnNative.cpp#L73

File: node/src/Context.cpp (hunk 3 above)
Comment (mingmingtasd) [reply node/src/Context.cpp:41]: OK, thanks, I will remove the lowercase.

File: node/src/Context.cpp (hunk 1 above)
Comment (mingmingtasd) [reply node/src/Context.cpp]: I will use switch and assert. Thanks! @fujunwei

File: node/src/Context.cpp [hunk 5]
```diff
@@ -71,7 +70,8 @@ namespace node {
 
         WebnnProcTable backendProcs = webnn_native::GetProcs();
         webnnProcSetProcs(&backendProcs);
-        mImpl = ml::Context::Acquire(webnn_native::CreateContext(&options));
+        mImpl = ml::Context::Acquire(
+            webnn_native::CreateContext(reinterpret_cast<MLContextOptions const*>(&options)));
```
Comment (fujunwei) [inline node/src/Context.cpp:74]: It's C API, so it's better to keep current implemenation and set the default value to MLContextOptions like
`MLContextOptions options = {MLPowerPreference_Default,  MLDevicePreference_Default};`

Comment (mingmingtasd) [reply node/src/Context.cpp:74]: OK, thanks!

**Review by fujunwei — APPROVED** [review]
LGTM, thanks.

**Review by huningxin — APPROVED** [review]
LGTM.

**Pull request merged**